source/lex.tex

```diff
@@ -256,8 +256,8 @@
 If a \grammarterm{universal-character-name} outside
 the \grammarterm{c-char-sequence}, \grammarterm{s-char-sequence}, or
 \grammarterm{r-char-sequence} of
-a character or
-string literal corresponds to a control character or
+a \grammarterm{character-literal} or
+\grammarterm{string-literal} corresponds to a control character or
```

Comment:
This change is unclear in the presence of UDLs. (*string-literal* is both a syntactic production and a morphological production, and it's unclear what this means.)

Reply:
We are in [lex], so grammar is assumed to refer to the lexical grammar. And UDLs do use the productions string-literal and character-literal.

Reply:
I don't think this is clear. (Similar questions: do the reservations in [lex.name]/3 apply to *ud-suffix*es, or only to identifier tokens? Is "orbit" as an identifier ill-formed because it contains the identifier "or" which is reserved and "shall not be used otherwise"?)

Adding an "(including within a *user-defined-literal*)" or similar would make this completely clear. If you think this is already clear normatively, maybe use a footnote "This includes a *string-literal* or *character-literal* within a *user-defined-string-literal* or *user-defined-character-literal* token." rather than a parenthetical?

Reply:
I don't think that [lex] is clear normatively, at all, in particular the little lexer sub-grammar used here with unclear relation to the "big" grammar.

Reply:
Fixed.